src/vector/__init__.py

```diff
@@ -96,6 +97,17 @@ def _import_awkward() -> None:
     )
 
 
+pytree: ModuleType | None = None
```

Comment:
The problem is we would like to find some way to tell the user that `optree` needs to be installed only if they attempt to access the `vector.pytree` module. We'd be better off just making a `src/vector/pytree.py` module so it can raise if not available, but then we'd have to access using `from vector.pytree import pytree`. Maybe that's not so bad?

Reply:
How about
```python
import vector

pytree = vector.register_pytree()
```
?

Reply:
Another advantage is then the docstring for this function would be a good place to describe what it is for.

Though, a module docstring for `vector.pytree` could also work